github/PullRequest.py

```diff
@@ -490,8 +490,12 @@ def create_review(
         assert is_optional(body, str), body
         assert is_optional(event, str), event
         assert is_optional_list(comments, dict), comments
-        post_parameters: dict[str, Any] = NotSet.remove_unset_items({"body": body})
-        post_parameters["event"] = "COMMENT" if is_undefined(event) else event
+        post_parameters: dict[str, Any] = NotSet.remove_unset_items({
+            "body": body,
+            "event": event,
+            "commit_id": commit.sha if is_defined(commit) else None,
```

Comment:
I think this should read `NotSet` to work:
```suggestion
            "commit_id": commit.sha if is_defined(commit) else NotSet,
```